src/domain/global_variable_view.rs

```diff
@@ -0,0 +1,383 @@
+                TypeEntryKind::TypeDef {
+                    name: type_name,
+                    type_ref,
+                } => {
+                    let global_variable = GlobalVariable {
+                        address: global_variable.address,
+                        name: global_variable.name,
+                        type_ref: type_ref.clone(),
+                    };
+
+                    self.from_global_variable(global_variable)
+                        .set_type_view(TypeView::TypeDef {
+                            name: type_name.clone(),
+                        })
+                }
+                TypeEntryKind::ConstType { type_ref } => {
+                    let global_variable = GlobalVariable {
+                        address: global_variable.address,
+                        name: global_variable.name,
+                        type_ref: type_ref.clone(),
+                    };
+
+                    self.from_global_variable(global_variable)
+                        .map_type_view(|type_view| TypeView::Const {
+                            type_view: Box::new(type_view),
+                        })
+                }
+                TypeEntryKind::PointerType { size, type_ref } => match type_ref {
+                    None => GlobalVariableView {
+                        name: global_variable.name.clone(),
+                        address: global_variable.address.clone(),
+                        size: *size,
+                        type_view: TypeView::VoidPointer,
+                        children: Vec::new(),
+                    },
+                    Some(type_ref) => GlobalVariableView {
+                        name: global_variable.name.clone(),
+                        address: global_variable.address.clone(),
+                        size: *size,
+                        type_view: TypeView::Pointer {
+                            type_view: Box::new(self.type_view_from_type_entry(type_ref)),
+                        },
+                        children: Vec::new(),
+                    },
+                },
+                TypeEntryKind::BaseType { name, size } => GlobalVariableView {
+                    name: global_variable.name.clone(),
+                    address: global_variable.address.clone(),
+                    size: *size,
+                    type_view: TypeView::Base { name: name.clone() },
+                    children: Vec::new(),
+                },
+                TypeEntryKind::StructureType { name, members } => {
+                    let base_address = &mut global_variable.address.clone();
+                    let members: Vec<GlobalVariableView> = members
+                        .iter()
+                        .map(|member| {
+                            self.from_structure_type_member_entry(member, base_address)
+                                .map_name(|member_name| {
+                                    format!("{}.{}", global_variable.name, member_name)
```

Comment:
This is a concern for the presentation.
The member should have an only member name and not have the parent name, like a.b.

Reply:
done